targetmem.h

```diff
@@ -258,7 +258,7 @@ data_to_val_aux (const matches_and_old_values_swath *swath,
                  size_t index, size_t swath_length)
 {
     unsigned int i;
-    value_t val;
+    volatile value_t val;
```

Comment:
I don't see this one when using `-Wclobbered`, how do you get it?

Reply:
Because value_t val is a struct type.
We don't know how compiler return/pass struct after optimization.
May through stack, register or memory access. Which depends on compiler decide really inline this function or not, and the usage of register.

According C standard, inline keyword for function which only recommend compiler inline the function. Not guaranteed compiler always inline the function.
Unless use this syntax `inline __attribute__((always_inline))`

In the other hand, pass/return struct usually gen more then one instruction. Once interrupt between instructions, struct data may incomplete which results unpredictable behavior.

[GCC Aggregate Return Impl](https://gcc.gnu.org/onlinedocs/gcc-3.3.6/gccint/Aggregate-Return.html)
In C99 the implantation of inline not defined very strict. And also GCC in different version or optimization level may have different behavior. 
[Inline Functions In C](https://www.greenend.org.uk/rjk/tech/inline.html)

So, basically return/pass struct in inline function without atomic access is a bad idea.
`volatile` here basically tells compiler not optimize and try its best to keep access atomically.